lock/etcd_test.go

```diff
@@ -32,6 +32,28 @@ type testEtcdClient struct {
 	err     error
 	putResp *client.PutResponse
 	getResp *client.GetResponse
+	txn     *testTxn
+}
+
+// testTxn implements the client.Txn interface
+type testTxn struct {
+	err   error
+}
+
+func (t *testTxn) If(cs ...client.Cmp) client.Txn {
```

Comment:
Maybe if defined like this it won't have to be a pointer in a struct and it's zero-value can be used, so you don't have to initialize it in all tests?
```suggestion
func (t testTxn) If(cs ...client.Cmp) client.Txn {
```